mlir/lib/Conversion/MIOpenToGPU/MIOpenToGPU.cpp

```diff
@@ -350,7 +350,7 @@ void LowerMIOpenOpsWithinGPUModulePass::runOnOperation() {
   populateAffineToStdConversionPatterns(patterns, &getContext());
   populateLoopToStdConversionPatterns(patterns, &getContext());
 
-  applyPatternsAndFoldGreedily(getOperation(), std::move(patterns));
+  (void)applyPatternsAndFoldGreedily(getOperation(), std::move(patterns));
```

Comment:
I think we should not suppress the return value but try to give us more information should it failed. I suggest either use:
 - llvm::errs() << "LowerMIOpenOpsWithinGPUModulePass failed to apply";
 - LLVM_DEBUG(llvm::dbgs() << "LowerMIOpenOpsWithinGPUModulePass failed to apply");

This apply to below `(void)` cases.

Reply:
This particular one MUST be suppressed and avoid early abortion of the pass. Otherwise tremendous amount of unit test would fail.

BTW it's incorrect to use `llvm::errs()` here.

Reply:
The most proper way is to invoke `signalPassFailure()`. Let me try add them in places suitable. There are places where we have to explicitly disregard the result.

Reply:
I'm incorrect here. It's the one at `mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp` which has to be suppressed, not this one.